providers/src/airflow/providers/google/cloud/hooks/apache_kafka.py

```diff
@@ -0,0 +1,289 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains a Managed Service for Apache Kafka hook."""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from copy import deepcopy
+from typing import TYPE_CHECKING
+
+from google.api_core.gapic_v1.method import DEFAULT, _MethodDefault
+from google.cloud.managedkafka_v1 import Cluster, ManagedKafkaClient, types
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.consts import CLIENT_INFO
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+
+if TYPE_CHECKING:
+    from google.api_core.operation import Operation
+    from google.api_core.retry import Retry
+    from google.cloud.managedkafka_v1.services.managed_kafka.pagers import ListClustersPager
+    from google.protobuf.field_mask_pb2 import FieldMask
+
+
+class ApacheKafkaHook(GoogleBaseHook):
```

Comment:
I have some concerns about the naming. This can be very very confusing with the apache kafka integration in the apache provider.
Is this the actual name Google use to lunch the service?

Reply:
@eladkal no, for client library the use name `managed_kafka` and the service in the GCP has name [Google Cloud Managed Service for Apache Kafka](https://cloud.google.com/products/managed-service-for-apache-kafka). I will change names for hook and operators from `ApacheKafka*` to `ManagedKafka*`. What do you think about this naming?